doc/api/esm.md

```diff
@@ -1096,20 +1099,185 @@ export async function resolve(specifier,
 With this loader, running:
 
 ```console
-NODE_OPTIONS='--experimental-loader ./custom-loader.mjs' node x.js
+NODE_OPTIONS='--experimental-loader ./custom-loader.mjs' node main.js
 ```
 
-would load the module `x.js` as an ES module with relative resolution support
+would load the module `main.js` as an ES module with relative resolution support
 (with `node_modules` loading skipped in this example).
 
+### Get source hook
+
+> Note: The loaders API is being redesigned. This hook may disappear or its
+> signature may change. Do not rely on the API described below.
+
+The `getFormat` hook provides a way to define a custom method of determining
+how a url should be interpreted. Using this hook will cause Node.js to
+ignore any format returned from `resolve`.
+
+```js
+const formatMap = {
+  'text/javascript': 'module',
+  'application/javascript': 'module'
+};
+/**
+ * @param {{url}} parameters
+ * @returns {{format}}
+ */
+export async function getFormat({ url, isMain, defaultGetFormat }) {
```

Comment:
Would you be open to making this signature `getFormat({ url, isMain }, defaultGetFormat)`?

I generally prefer the pattern of having the parent function as the second argument as it is part of the hook model itself. Also for consistency with getSource.

Reply:
seems ok, but a lot of these functions seem to be unable to call each other while we leave these as arguments to each function. I'd prefer we bundle all the delegation calls into a single object eventually.